### ISSUE TITLE
defmethod grammar comments wrong

### ISSUE THREAD

**Issue by bkerin** [CObjectSystem/COS#25]
The comments at the top of CosBase/include/cos/cos/method.h include:

```
   method-definition:
     defmethod( generic-specifier )
       method-statement
     endmethod
```

Where generic-specifier presumably refers to CosBase/include/cos/cos/generic.h where there is this:

```
   generic-specifier:
     return-type , generic-def , param-list

   generic-def:
     generic-name
     ( class-name ) generic-name
```

But when generic names appear in method definitions they specify around methods, not class names (and the word following the tuple is presumably being interpreted as a tag as mentioned in the OOPSLA paper).  As in CosStd/src/File.c:

```
defmethod(void, (ginitialize)StdioInit, Any)`
```

This looks like a case where the grammar in the user notes got over-factored, as the paper seems more correct.

I'll be happy to phrase all reports like this as pull requests if you like, but I'd hate to make things worse and I'm not yet confident of the correct fixes for problems like this one (where I don't know what tag names are for exactly).

I guess the most simple fix would be to add another entry to this list that appears in method.h:

```
   where in the context of methods:
     - the param-name_opt in param-decl can be omitted for unused argument.
     - the last param-decl must be (va_list)va for variadic generics.
```

But it's already somewhat confusing and it might be better to just write out in method.h the full grammar for the flavor of generic-specifier accepted in defmethod() context.

**Comment by ldeniau** [comment]
I agree that generic-specifier cannot be used for defmethod. the simplest is to refer to the paper (or copy-paste the paper there)
